src/docker-utils.coffee

```diff
@@ -62,46 +62,78 @@ do ->
 		.disposer (release) ->
 			release()
 
-	exports.rsyncImageWithProgress = (imgDest, { requestTimeout, totalTimeout, startFromEmpty = false }, onProgress) ->
+	exports.rsyncImageWithProgress = (imgDest, { requestTimeout, totalTimeout, uuid, apiKey, startFromEmpty = false }, onProgress) ->
 		Promise.using readLockImages(), ->
 			Promise.try ->
 				if startFromEmpty
 					return 'resin/scratch'
 				findSimilarImage(imgDest)
 			.then (imgSrc) ->
 				new Promise (resolve, reject) ->
-					progress request.get("#{config.deltaHost}/api/v2/delta?src=#{imgSrc}&dest=#{imgDest}", timeout: requestTimeout)
-					.on 'progress', (progress) ->
-						# In request-progress ^2.0.1, "percentage" is a ratio from 0 to 1
-						onProgress(percentage: progress.percentage * 100)
-					.on 'end', ->
-						onProgress(percentage: 100)
-					.on 'response', (res) ->
-						if res.statusCode is 504
-							reject(new Error('Delta server is still processing the delta, will retry'))
-						else if res.statusCode isnt 200
-							reject(new Error("Got #{res.statusCode} when requesting image from delta server."))
-						else
-							if imgSrc is 'resin/scratch'
-								deltaSrc = null
-							else
-								deltaSrc = imgSrc
-							res.pipe(dockerDelta.applyDelta(deltaSrc, imgDest))
-							.on('id', resolve)
-							.on('error', reject)
-					.on 'error', reject
+					docker.getRegistryAndName(imgDest)
+					.then ({ registry, imageName, tagName }) ->
+						tokenEndpoint = "#{config.apiEndpoint}/auth/v1/token"
+						authOpts =
+							auth:
+								user: 'd_' + uuid
+								pass: apiKey
+								sendImmediately: true
+						request.get "#{tokenEndpoint}?service=#{registry}&scope=repository:#{imageName}:pull", authOpts, (err, res, body) ->
```

Comment:
Will this also work to authenticate the delta server to pull `imgSrc`? As it will have to be able to pull both the source and destination to compute the delta